pkg/operator/controller/ingress/monitoring_test.go

```diff
@@ -41,7 +41,12 @@ func Test_serviceMonitorChanged(t *testing.T) {
 	}
 	if changed, sm3 := serviceMonitorChanged(sm1, sm2); !changed {
 		t.Fatal("expected changed to be true after clearing servicemonitor's selector")
-	} else if changedAgain, _ := serviceMonitorChanged(sm2, sm3); changedAgain {
-		t.Fatal("serviceMonitorChanged does not behave as a fixed-point function")
+	} else {
+		if updatedChanged, _ := serviceMonitorChanged(sm1, sm3); !updatedChanged {
+			t.Error("serviceMonitorChangedChanged reported changes but did not make any update")
```

Comment:
nit:
```suggestion
			t.Error("serviceMonitorChanged reported changes but did not make any update")
```

Reply:
Thanks!  Fixed in https://github.com/openshift/cluster-ingress-operator/compare/3a56b8dbad66a5ca58fdd7d140985872e119c173..3d06500a64b02ae95d533bebc03fd4bee019e05b.